### ISSUE TITLE
Travis CI issues?

### ISSUE THREAD

**Issue by mnlevy1981** [marbl-ecosys/MARBL#343]
First, it looks like test results from #342 were never actually reported in the PR. (I think I fixed this by revoking access to Travis CI and then reinstating it... will find out for sure with next PR).

Even worse, though, it looks like the tests actually [failed](https://travis-ci.org/github/marbl-ecosys/MARBL/builds/684377969). Python 2.7 wasn't able to get past the `pip install` and Python 3.6 was throwing exceptions in the netCDF comparison:

```
Traceback (most recent call last):
  File "./netcdf_comparison.py", line 274, in <module>
    ds_base_in, ds_new_in = _open_files(args.baseline, args.new_file) # pylint: disable=invalid-name
  File "./netcdf_comparison.py", line 66, in _open_files
    return xr.open_dataset(baseline), xr.open_dataset(new_file)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/xarray/backends/api.py", line 499, in open_dataset
    engine = _get_default_engine(filename_or_obj, allow_remote=True)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/xarray/backends/api.py", line 153, in _get_default_engine
    engine = _get_default_engine_netcdf()
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/xarray/backends/api.py", line 96, in _get_default_engine_netcdf
    import netCDF4  # noqa: F401
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/netCDF4/__init__.py", line 3, in <module>
    from ._netCDF4 import *
  File "netCDF4/_netCDF4.pyx", line 1213, in init netCDF4._netCDF4
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/cftime/__init__.py", line 1, in <module>
    from ._cftime import utime, JulianDayFromDate, DateFromJulianDay
  File "__init__.pxd", line 918, in init cftime._cftime
ValueError: numpy.ufunc size changed, may indicate binary incompatibility. Expected 216 from C header, got 192 from PyObject
```

The good news is that the failures all appear to be in the Travis environment, rather than indicative of actual issues with the code. I bet a fix for this is to pin versions of packages in the pip call.

A few related issues:

1. This isn't the first time I've had TravisCI stop testing a repository (possibly not even the first time it stopped testing **this** repository) and it's a little frustrating. Perhaps CircleCI would be more stable?
1. I don't know if we need to continue testing with Python 2.7; I'm happy to continue doing so until we've had a chance to discuss it, but I don't think anyone is still relying on python 2 at this point
1. I'd like to add python 3.7 (and maybe 3.8?) to the test suite. I don't expect any failures from those later versions, but if there ARE issues it would be nice to find out from continuous integration rather than surprising a user with an incompatibility

**Comment by mnlevy1981** [comment]
For 2.7, the last successful test got the following from `pip`:

```
PyYAML-5.3.tar.gz
xarray-0.11.3-py2.py3-none-any.whl
netCDF4-1.5.3-cp27-cp27mu-manylinux1_x86_64.whl
pandas-0.24.2-cp27-cp27mu-manylinux1_x86_64.whl
cftime-1.0.4.2-cp27-cp27mu-manylinux1_x86_64.whl
python_dateutil-2.8.1-py2.py3-none-any.whl
pytz-2019.3-py2.py3-none-any.whl
```

The failing build was trying to use the following:

```
PyYAML-5.3.1.tar.gz
xarray-0.11.3-py2.py3-none-any.whl
netCDF4-1.5.3-cp27-cp27mu-manylinux1_x86_64.whl
pandas-0.24.2-cp27-cp27mu-manylinux1_x86_64.whl
cftime-1.1.2.tar.gz
python_dateutil-2.8.1-py2.py3-none-any.whl
pytz-2020.1-py2.py3-none-any.whl
```

And the error was

```
Failed building wheel for cftime
```

So one possibility is pinning `cftime=1.0.4.2` in the `pip install`. That would likely cause problems in the 3.6 testing, where the last good install used `cftime-1.1.0-cp36-cp36m-manylinux1_x86_64.whl` and the failed test successfully installed `cftime-1.1.2-cp36-cp36m-manylinux1_x86_64.whl`.

I'll start with `cftime<1.1.2` and see how that goes.

**Comment by mnlevy1981** [comment]
From [stackoverflow](https://stackoverflow.com/questions/40845304/runtimewarning-numpy-dtype-size-changed-may-indicate-binary-incompatibility), it looks like the buggy version of `numpy` was 1.15.0... forcing `numpy>=1.16` fixes the 3.6 issue. Not sure what to do about building a python 2.7 environment, especially because I can't replicate this issue on my VM that was supposed to mimic the Travis environment.

**Comment by mnlevy1981** [comment]
Travis is working after #344, though it's limited to python 3.6 at the moment
